messaging/src/test/java/org/axonframework/eventhandling/deadletter/jpa/EventMessageDeadLetterJpaConverterTest.java

```diff
@@ -0,0 +1,198 @@
+import org.junit.jupiter.api.*;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class EventMessageDeadLetterJpaConverterTest {
+
+    private static final String PAYLOAD_REVISION = "23.0";
+    private final EventMessageDeadLetterJpaConverter converter = new EventMessageDeadLetterJpaConverter();
+    private final Serializer serializer = TestSerializer.JACKSON.getSerializer();
+    private final ConverterTestEvent event = new ConverterTestEvent("myValue");
+    private final MetaData metaData = MetaData.from(Collections.singletonMap("myMetadataKey", "myMetadataValue"));
+
+    @Test
+    void testCanConvertGenericEventMessageAndBackCorrectly() {
+        test(GenericEventMessage.asEventMessage(event).andMetaData(metaData));
+    }
+
+    @Test
+    void testCanConvertDomainEventMessageAndBackCorrectly() {
+        test(new GenericDomainEventMessage<>("MyType", "8239081092", 25L, event, metaData));
+    }
+
+    @Test
+    void testCanConvertTrackedDomainEventMessageWithGlobalSequenceTokenAndBackCorrectly() {
+        test(new GenericTrackedDomainEventMessage<>(new GlobalSequenceTrackingToken(232323L),
+                                                    "MyType",
+                                                    "8239081092",
+                                                    25L,
+                                                    new GenericEventMessage<>(event, metaData),
+                                                    Instant::now));
+    }
+
+    @Test
+    void testCanConvertTrackedDomainEventMessageWithGapAwareTokenAndBackCorrectly() {
+        test(new GenericTrackedDomainEventMessage<>(new GapAwareTrackingToken(232323L, Arrays.asList(24L, 255L, 2225L)),
+                                                    "MyType",
+                                                    "8239081092",
+                                                    25L,
+                                                    new GenericEventMessage<>(event, metaData),
+                                                    Instant::now));
+    }
+
+    @Test
+    void testCanConvertTrackedEventMessageWithGlobalSequenceTokenAndBackCorrectly() {
+        test(new GenericTrackedEventMessage<>(new GlobalSequenceTrackingToken(232323L),
+                                              new GenericEventMessage<>(event, metaData),
+                                              Instant::now));
+    }
+
+
+    @Test
+    void testCanConvertTrackedEventMessageWithGapAwareTokenAndBackCorrectly() {
+        test(new GenericTrackedEventMessage<>(new GapAwareTrackingToken(232323L, Arrays.asList(24L, 255L, 2225L)),
+                                              new GenericEventMessage<>(event, metaData),
+                                              Instant::now));
+    }
+
+    private void test(EventMessage<?> message) {
```

Comment:
Minor, but changing the methog name to something like 'testConversion' would make the test more readable, I think.